components/webui/client/src/components/SqlEditor/index.tsx

```diff
@@ -0,0 +1,142 @@
+                // accepts the first suggestion. To prevent accidental auto-completion
+                // in multi-line queries and to allow users to dismiss suggestions more easily,
+                // we make the current input the first suggestion.
+                // Users can then use arrow keys to select a keyword if needed.
+                const typedWord = model.getValueInRange(range);
+                if (0 < typedWord.length) {
+                    suggestions.push({
+                        detail: "Current",
+                        insertText: `${typedWord}\n`,
+                        kind: monaco.languages.CompletionItemKind.Text,
+                        label: typedWord,
+                        range: range,
+                    });
+                }
+
+                return {
+                    suggestions: suggestions,
+                    incomplete: true,
+                };
+            },
+            triggerCharacters: [
+                " ",
+                "\n",
+            ],
+        });
+
+        return () => {
+            provider.dispose();
+        };
+    }, [monacoEditor]);
+
+    const [isContentMultiline, setIsContentMultiline] = useState<boolean>(false);
+
+    const handleMonacoMount = useCallback((editor: monaco.editor.IStandaloneCodeEditor) => {
+        editor.onDidContentSizeChange((ev) => {
+            if (false === ev.contentHeightChanged) {
+                return;
+            }
+            if (null === monacoEditor) {
+                throw new Error("Unexpected null Monaco instance");
+            }
+            const domNode = editor.getDomNode();
+            if (null === domNode) {
+                throw new Error("Unexpected null editor DOM node");
+            }
+            const model = editor.getModel();
+            if (null === model) {
+                throw new Error("Unexpected null editor model");
+            }
+            const lineHeight = editor.getOption(monacoEditor.editor.EditorOption.lineHeight);
+            const contentHeight = editor.getContentHeight();
+            const approxWrappedLines = Math.round(contentHeight / lineHeight);
+            setIsContentMultiline(1 < approxWrappedLines);
+            if (MAX_VISIBLE_LINES >= approxWrappedLines) {
+                domNode.style.height = `${contentHeight}px`;
+            } else {
+                domNode.style.height = `${lineHeight * MAX_VISIBLE_LINES}px`;
+            }
+        });
+    }, [monacoEditor]);
+
+    return (
+        <Editor
+            language={"sql"}
+
+            // Use white background while loading (default is grey) so transition to editor with
+            // white background is less jarring.
+            loading={<div style={{backgroundColor: "white", height: "100%", width: "100%"}}/>}
+            options={{
+                automaticLayout: true,
+                folding: isContentMultiline,
+                fontSize: 18,
```

Comment:
@hoophalab we discussed offline that setting the `fontSize` to 30 could make the UI look awkward. Still, i think it makes sense to let the input box's height to match the Run button's height. how about:
```suggestion
                fontSize: 20,
                lineHeight: 30,
```